src/drivers/cmakefileapi/api_helpers.ts

```diff
@@ -0,0 +1,244 @@
+import * as api from '@cmt/api';
+import * as cache from '@cmt/cache';
+import * as index_api from '@cmt/drivers/cmakefileapi/api';
+import {
+  CodeModelConfiguration,
+  CodeModelContent,
+  CodeModelFileGroup,
+  CodeModelProject,
+  CodeModelTarget
+} from '@cmt/drivers/codemodel-driver-interface';
+import * as logging from '@cmt/logging';
+import {fs} from '@cmt/pr';
+import rollbar from '@cmt/rollbar';
+import * as path from 'path';
+
+const log = logging.createLogger('cmakefileapi-helper');
+
+export async function createQueryFileForApi(api_path: string): Promise<string> {
+  const query_path = path.join(api_path, 'query', 'client-vscode');
+  const query_file_path = path.join(query_path, 'query.json');
+  await fs.mkdir_p(query_path);
+
+  const requests
+      = {requests: [{kind: 'cache', version: 2}, {kind: 'codemodel', version: 2}, {kind: 'cmakeFiles', version: 1}]};
+
+  await fs.writeFile(query_file_path, JSON.stringify(requests));
+  return query_file_path;
+}
+
+export async function loadIndexFile(reply_path: string): Promise<index_api.Index.IndexFile|null> {
+  log.debug(`Read reply folder: ${reply_path}`);
+  if (!await fs.exists(reply_path)) {
+    return null;
+  }
+
+  const files = await fs.readdir(reply_path);
+  log.debug(`Found index files: ${JSON.stringify(files)}`);
+
+  const index_file = files.find(filename => filename.startsWith('index-'));
```

Comment:
The spec says that it is possible that multiple index files can exist at the same time.  Can we check and make sure we get the latest?  They will be sorted lexicographically.

Reply:
I believe this is a duplicate.  I wrote it again because github.com wasn't showing me the preview.